### PR TITLE
[ABM-Edge-7] Implement initial stubs for the UI 

### PR DESCRIPTION
**Pull request by Shabirmean**
### Depends on:
- Please review _(or make sure **merged** status of)_ the following PR before starting on this: 
   - #6 

### What's this?
- Implement the js for consuming the api-server APIs from the UI source
- Make minor changes to the api-server APIs to work with the UI calls
- Re-arrange the `pom.xmls` to ensure that all the necessary modules are part of the build process 

### Architecture
image

***Notes on the Architecture:***
- The initial MVP will implement the data-store in memory. We will work on the expanding the app with an external storage option after the MVP
- The app exposes a _/switch_ functionality to switch between store types
- More details at: [go/abm-edge-app](http://go/abm-edge-app)

### CODE REVIEW

File: point-of-sale-app/ui/src/services/api/RetailEdge.js [hunk 1]
```diff
@@ -14,53 +14,40 @@
...
-        ...HEADERS,
-        'Authorization': `Bearer ${store.getters.accessToken}`,
-      }
-    }).catch((error) => {
-      if (!error.response) {
-        return { status: 500 };
-      }
-    });
+  async types() {
+    return get(`${API_SERVER_URL}/api/types`, HEADERS);
```
Comment (NimJay) [inline point-of-sale-app/ui/src/services/api/RetailEdge.js:51]: Suggestion: If you think we won't be adding any more files into `/ui/src/services/` (even in the future), I would remove sub-folders `services/api/` and `services/lib/`, and just place `RetailEdge.js` and `Util.js` inside `services/`. Feels like a lot of nesting.

**Review by NimJay — APPROVED** [review]
I just took a look at the `ui` code. I like the simplicity.
I left a few comments — nothing urgent so feel free to ignore some of my comments.
Awesome work, Shabir! :)

File: point-of-sale-app/ui/src/services/api/RetailEdge.js (hunk 1 above)
Comment (Shabirmean) [reply point-of-sale-app/ui/src/services/api/RetailEdge.js]: Removed nesting in https://github.com/GoogleCloudPlatform/anthos-edge-usecases/pull/7/commits/183c935b6f9036222e8fae0ae8f25ce1915a4070

File: point-of-sale-app/ui/src/main.js [hunk 2]
```diff
@@ -20,19 +20,21 @@ import App from './App.vue'
...
 import '@fortawesome/fontawesome-free/js/all.js'
 import './css/app.css'
+import 'vue-toast-notification/dist/theme-default.css';
 
 Vue.use(Vuex)
 Vue.use(BootstrapVue)
 Vue.use(AudioRecorder)
+// https://vuejsexamples.com/yet-another-toast-notification-plugin-for-vue-js/
+Vue.use(VueToast);
 Vue.mixin(titleMixin)
```
[the remark this answers is not in the document]
Comment (Shabirmean) [reply point-of-sale-app/ui/src/main.js]: Fixed semicolon issues in: https://github.com/GoogleCloudPlatform/anthos-edge-usecases/pull/7/commits/f6dd520c0d107c27ab12cd0ba515673da94049fa

File: point-of-sale-app/ui/README.md [hunk 3]
```diff
@@ -1,3 +1,34 @@
 # UI
 
 - Anthos BareMetal Edge Workload UI
+### Local development of the UI
+
+***The following steps assume that you are running them form the root of this repository***
```
[the remark this answers is not in the document]
Comment (Shabirmean) [reply point-of-sale-app/ui/README.md:6]: Fixed numbering in: https://github.com/GoogleCloudPlatform/anthos-edge-usecases/pull/7/commits/bc2189b1d0c6842e912b49a6f1cd5dba6ab8878e

File: point-of-sale-app/ui/package.json [hunk 4]
```diff
@@ -14,11 +14,16 @@
     "core-js": "^3.6.5",
     "firebase": "^8.2.4",
     "firebaseui": "^4.7.1",
+    "mosha-vue-toastify": "^1.0.20",
```
[the remark this answers is not in the document]
Comment (Shabirmean) [reply point-of-sale-app/ui/package.json]: Removed all unused dependencies in https://github.com/GoogleCloudPlatform/anthos-edge-usecases/pull/7/commits/a18eeba3aa1646abd71c4c0325f769884206cfa6

File: point-of-sale-app/ui/src/main.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shabirmean) [reply point-of-sale-app/ui/src/main.js]: Added google style guide for linting: https://github.com/GoogleCloudPlatform/anthos-edge-usecases/pull/7/commits/92d81f85202806a46407635d24866dfa034a18b2

[the remark this answers is not in the document]
Comment (NimJay) [reply point-of-sale-app/ui/src/main.js]: So glad you added the Google style guide! Thanks, Shabir.

**Review by puneith — APPROVED** [review]
lgtm

**Review by puneith — APPROVED** [review]
LGTM

**Review by NimJay — APPROVED** [review]
Sweet, we're using Artifact Registry now.
I skimmed the changes quickly. LGTM!

**Pull request merged**